ipatests/test_integration/test_ipa_ipa_migration.py

```diff
@@ -1265,3 +1265,44 @@ def test_check_ad_idrange_is_migrated(self):
             ["ipa", "idrange-show", ad_domain_name + "_id_range"]
         )
         assert cmd1.stdout_text == cmd2.stdout_text
+
+
+class TestIPAMigratewithBackupRestore(IntegrationTest):
+    """
+    Tests for ipa-migrate tool with backup files
+    """
+    num_replicas = 1
+    topology = "line"
+
+    def test_ipa_migrate_stage_mode(self):
+        """
+        This test checks ipa-migrate with LDIF file
+        from backup of remote server is successful.
+        """
+        ERR_MSG = (
+            "error: change collided with another change"
+        )
+        dashed_domain_name = self.master.domain.realm.replace(
+            ".", '-'
+        )
+        user_root_ldif_file = '{}-userRoot.ldif'.format(
+            dashed_domain_name
+        )
+        param = ['-n','-f', user_root_ldif_file]
+        tasks.kinit_admin(self.master)
+        tasks.kinit_admin(self.replicas[0])
+        backup_path = tasks.get_backup_dir(self.master)
+        ipa_tar_file = backup_path + '/ipa-full.tar'
+        self.master.run_command(
+            ['/usr/bin/tar', '-xvf', ipa_tar_file]
+        )
+        result = run_migrate(
+            self.replicas[0],
+            "stage-mode",
+            self.master,
```

Comment:
self.master is a ipatests.pytest_ipa.integration.host.Host object. Replace with `self.master.hostname` if you need the host name.